src/lua/serializer.c

```diff
@@ -76,12 +76,32 @@ static struct {
 	OPTION(LUA_TBOOLEAN, encode_use_tostring, 0),
 	OPTION(LUA_TBOOLEAN, encode_invalid_as_nil, 0),
 	OPTION(LUA_TBOOLEAN, encode_error_as_ext, 1),
+	OPTION(LUA_TBOOLEAN, encode_esc_slash, 0),
 	OPTION(LUA_TBOOLEAN, decode_invalid_numbers, 1),
 	OPTION(LUA_TBOOLEAN, decode_save_metatables, 1),
 	OPTION(LUA_TNUMBER,  decode_max_depth, 128),
 	{ NULL, 0, 0, 0},
 };
 
+/** toggler for json.encode behavior change with tarantool.compat (gh-7000) */
+int
+json_esc_slash_toggle(struct lua_State *L)
+{
+	assert(lua_isboolean(L, -1) && "boolean argument expected");
+	bool value = lua_toboolean(L, -1);
+	lua_pop(L, 1);
+
+	for (int i = 0; OPTIONS[i].name != NULL; i++) {
+		if (!strcmp(OPTIONS[i].name, "encode_esc_slash")) {
+			OPTIONS[i].defvalue = value;
+			lua_pushnumber(L, 0);
+			return 1;
```

Comment:
Why we need to push zero?

Reply:
Sure, it means `return N`, not pushing a value. IOW, you can replace

```c
lua_pushnumber(L, 0);
return 1;
```

with

```c
return 0;
```

The function will push zero values to the Lua stack (will return nothing from a Lua caller perspective).

Reply:
It is fixed now.